### DISCUSSION TITLE
Spoilers?

### DISCUSSION THREAD

**Discussion by SirWumpus** [ioccc-src/temp-test-ioccc#2299] [General]
How(e) will spoilers be handled?  Submitted by winners post judging?  Or pre-bundled as a tarball or password encrypted archive of some sort?

**Comment by lcn2** [comment]
The IOCCC today is focused on the understanding a winning entries rather than keeping the internals a _mystery_.  The days where IOCCC treated even the summary of an entry as a spoiler are long gone.   Historic explanations that were once "rot13-ed" have been de-rotated, often with a heading of "spoiler alert" above the text.  And there will no longer be a "years-spoiler.html" page on the new site. 

People will still be free to just download the entry's compressed tarball and not read the `README.md` (nor view any `index.html` web page) (if they wish).  This has been made easier for people who prefer to just look at the `prog.c` file (if they wish), run `make` (if they wish), and run the often used `try.sh` script (if they wish) so they don't even need to read anything about how to run the entry.  For those who prefer to not read anything, we have made that process easier.   However, even though we plan to accommodate those who prefer to read nothing about the entry, such people are **NOT** the primary target audience for the IOCCC.

The IOCCC prefers (and will mention in the new guidelines), that fs an author wishes to put detailed information about an entry "to the side", then we recommend the model used by [2015/burton](https://ioccc-src.github.io/temp-test-ioccc/2015/burton/index.html) where such side content is put into a separate markdown page say _details.md_or even _spoiler.md_) and link to that separate markdown page on your remarks with a suitable so-called "spoiler warning".

Probably a FAQ about this topic is in order.

Again, the emphasis is on understanding entries and gaining new insights into the C language are what matters now.  Detailed explanations about the entry by the author are preferred and add value to the entry.  It is one of the reasons why _one liner_ and/or _short  source code_ are still valued by the IOCCC judges: because they offer a batter opportunity for even the novice programmer to fully dig into an entry.  We **DO VALUE** larger source code (right up to the Rule 2 size limit), as evidenced by the fast that most IOCCC winners are larger.

A good explanation by the author about insights into their entry adds **significant value**  to their entry.  And if the author prefers to move much of that explanation into the side markdown file while giving someone the option to read it later on, that would be appreciated.

**Comment by lcn2** [comment]
> How(e) will spoilers be handled? Submitted by winners post judging? Or pre-bundled as a tarball or password encrypted archive of some sort?

A detailed explanation about an entry can be put into markdown side file, with a link from the author's remarks.  They can put a so-called "spoiler alert" if they wish, while linking to that markdown side file.  They can even call the markdown side file _spoiler.md_ if they wish.  

The author can, in the author's remarks file, "tease" the reader by asking them pointed questions or challenges about how their code works.  They can admonish the reader to not read the so-called "spoiler alert" until after the reader as made an effort to answer such questions.

Here again, an educational / tutorial approach over keeping it a "mystery" is preferred.

**Reply by SirWumpus** [reply]
All valid points WRT spoiler documentation.  I was more interested in how the unobfuscated original source code that an entry may be derived from might be presented, eg.  https://www.ioccc.org/2015/howe/spoiler/prog.c.  Should it come with the submission or only be supplied (if they desire) by the author only if they win?
